src/Microsoft.DocAsCode.Build.Common/ValidateBookmark.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.Common
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.IO;
+    using System.Linq;
+    using System.Text;
+    using System.Web;
+
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.Plugins;
+    using Microsoft.DocAsCode.Utility;
+    using HtmlAgilityPack;
+
+    [Export(nameof(ValidateBookmark), typeof(IPostProcessor))]
+    public class ValidateBookmark : IPostProcessor
+    {
+        private static readonly string XpathTemplate = "//*/@{0}";
+
+        public ImmutableDictionary<string, object> PrepareMetadata(ImmutableDictionary<string, object> metadata)
+        {
+            return metadata;
+        }
+
+        public Manifest Process(Manifest manifest, string outputFolder)
+        {
+            if (outputFolder == null)
+            {
+                throw new ArgumentNullException("Base directory can not be null");
+            }
+            var registeredBookmarks = new Dictionary<string, HashSet<string>>(new FilePathComparer());
+            var bookmarks = new Dictionary<string, List<Tuple<string, string>>>(new FilePathComparer());
+            var fileMapping = new Dictionary<string, string>(new FilePathComparer());
+
+            foreach (var p in from item in manifest.Files ?? Enumerable.Empty<ManifestItem>()
+                                         from output in item.OutputFiles
+                                         where output.Key.Equals(".html", StringComparison.OrdinalIgnoreCase)
+                                         select new
+                                         {
+                                             RelativePath = output.Value.RelativePath,
+                                             SrcRelativePath = item.SourceRelativePath,
+                                         })
+            {
+                string srcRelativePath = p.SrcRelativePath;
+                string relativePath = p.RelativePath;
+                var filePath = Path.Combine(outputFolder, relativePath);
+                fileMapping[relativePath] = srcRelativePath;
+                var html = new HtmlDocument();
+
+                if (File.Exists(filePath))
+                {
+                    try
+                    {
+                        html.Load(filePath, Encoding.UTF8);
+                    }
+                    catch (Exception ex)
+                    {
+                        Logger.LogWarning($"Warning: Can't load content from {filePath}: {ex.Message}");
+                        continue;
+                    }
+
+                    var links = GetNodeAttribute(html, "src").Concat(GetNodeAttribute(html, "href"));
```

Comment:
As the last post processor, the html may contains <link href="abc" /> or <script src="abc" />, should we check src and href in "a" tag only or all of them?